src/graphql/resolvers/ssb.js

```
@@ -42,6 +123,110 @@ module.exports = function Resolvers () {
+   * @param {int} opts.threadMaxSize - max amount of messages in each thread to return
+   */
+  const getThreads = (id, opts) => {
+    const { threadMaxSize, limit = 10 } = opts
+    return new Promise((resolve, reject) => {
+      pull(
+        ssb.threads.profile({ id, reverse: true, threadMaxSize }),
+        pull.take(limit),
+        pull.collect((err, threads) => {
+          if (err) return reject(err)
+
+          const res = threads.map(({ messages }) => {
+            return {
+              id: messages[0].key,
+              messages
+            }
+          })
+
+          resolve(res)
+        })
+      )
+    })
+  }
+
+  /**
+   * Takes the messages from a thread and maps them based on whether
+   * a profile is returned for the author of that message or not.
+   * Messages from someone who hasnt yet opted in to publicWebHosting or
+   * a profile wasnt found for them, will return empty values
+   * @param {array} messages - messages in a thread
+   */
+  const mapMessages = (messages) => {
+    return new Promise((resolve, reject) => {
+      pull(
+        pull.values(messages),
+        pullParaMap((msg, cb) => {
+          getProfile(msg.value.author)
+            .then(profile => {
+              // if there was a profile, thats great
+              if (profile) {
+                return cb(null, {
+                  id: msg.key,
+                  author: msg.value.author,
+                  text: (msg.value.content.text || ''),
+                  timestamp: msg.value.timestamp // asserted publish time
+                })
+              }
+
+              // if their isnt (could be they havent yet opted in to publicWebHosting)
+              // we need to return an empty msg instead
+              cb(null, {
+                id: null,
+                text: null,
+                timestamp: msg.timestamp,
+                author: null
+              })
+            })
+            .catch(err => cb(err))
```

I think the Error case should fallback to hiding the content:

```js
         .catch(err => {
            cb(null, {
                id: null,
                text: null,
                timestamp: msg.timestamp,
                author: null
            })
         })
```